gsitemap.php

```diff
@@ -1,907 +1,987 @@
+    {
+        if ($i <= 25000 && memory_get_usage() < 100000000) {
+            $link_sitemap[] = $new_link;
+            $i++;
+
+            return true;
+        } else {
+            $this->_recursiveSitemapCreator($link_sitemap, $lang, $index);
+            if ($index % 20 == 0 && !$this->cron) {
+                $this->context->smarty->assign(
+                    array(
+                        'gsitemap_number' => (int)$index,
+                        'gsitemap_refresh_page' => './index.php?tab=AdminModules&configure=gsitemap&token='.Tools::getAdminTokenLite('AdminModules').'&tab_module='.$this->tab.'&module_name=gsitemap&continue=1&type='.$new_link['type'].'&lang='.$lang.'&index='.$index.'&id='.(int)$id_obj.'&id_shop='.$this->context->shop->id
+                    )
+                );
+
+                return false;
+            } elseif ($index % 20 == 0 && $this->cron) {
+                header('Refresh: 5; url=http'.(Configuration::get('PS_SSL_ENABLED') ? 's' : '').'://'.Tools::getShopDomain(false, true).__PS_BASE_URI__.'modules/gsitemap/gsitemap-cron.php?continue=1&token='.Tools::substr(Tools::encrypt('gsitemap/cron'), 0, 10).'&type='.$new_link['type'].'&lang='.$lang.'&index='.$index.'&id='.(int)$id_obj.'&id_shop='.$this->context->shop->id);
+                die();
+            } else {
+                if ($this->cron) {
+                    header('location: http'.(Configuration::get('PS_SSL_ENABLED') ? 's' : '').'://'.Tools::getShopDomain(false, true).__PS_BASE_URI__.'modules/gsitemap/gsitemap-cron.php?continue=1&token='.Tools::substr(Tools::encrypt('gsitemap/cron'), 0, 10).'&type='.$new_link['type'].'&lang='.$lang.'&index='.$index.'&id='.(int)$id_obj.'&id_shop='.$this->context->shop->id);
+                } else {
+                    $admin_folder = str_replace(_PS_ROOT_DIR_, '', _PS_ADMIN_DIR_);
+                    $admin_folder = Tools::substr($admin_folder, 1);
+                    header('location: http'.(Configuration::get('PS_SSL_ENABLED') ? 's' : '').'://'.Tools::getShopDomain(false, true).__PS_BASE_URI__.$admin_folder.'/index.php?tab=AdminModules&configure=gsitemap&token='.Tools::getAdminTokenLite('AdminModules').'&tab_module='.$this->tab.'&module_name=gsitemap&continue=1&type='.$new_link['type'].'&lang='.$lang.'&index='.$index.'&id='.(int)$id_obj.'&id_shop='.$this->context->shop->id);
+                }
+                die();
+            }
+        }
+    }
+
+    /**
+     * Hydrate $link_sitemap with home link
+     *
+     * @param array  $link_sitemap contain all the links for the Google Sitemap file to be generated
+     * @param string $lang         language of link to add
+     * @param int    $index        index of the current Google Sitemap file
+     * @param int    $i            count of elements added to sitemap main array
+     *
+     * @return bool
+     */
+    protected function _getHomeLink(&$link_sitemap, $lang, &$index, &$i)
+    {
+        $link = new Link();
```

Comment:
You should be able to use `$this->context->link` instead.

Reply:
Fixed

Reply:
Is it?

Reply:
It was, I rolled back.

Using <code>$this->context->link</code> does not include protocol. Let me look into this.

Reply:
Can you help me, how exactly do you want the change to be?